### PR TITLE
Mosip-32302 Move develop resident service releated code changes to release 121x

### PR DESCRIPTION
**Pull request by kameshsr**

### CODE REVIEW

File: db_upgrade_scripts/mosip_idrepo/sql/1.2.0.1_to_1.2.1.0_upgrade.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+\echo 'Upgrade Queries not required for transition from $CURRENT_VERSION to $UPGRADE_VERSION'
```
Comment (LoganathanSekar7627) [inline db_upgrade_scripts/mosip_idrepo/sql/1.2.0.1_to_1.2.1.0_upgrade.sql:1]: identity_update_count_tracker  table is added. Please add the DB scripts for that.

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/provider/IdentityUpdateTrackerPolicyProvider.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.mosip.idrepository.core.util.EnvUtil;
+
+/**
+ * @author Manoj SP
+ * @since 1.2.0.2
```
Comment (LoganathanSekar7627) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/provider/IdentityUpdateTrackerPolicyProvider.java:21]: Make it @ssince 1.2.1.0

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java [hunk 3]
```diff
@@ -745,9 +860,6 @@ private void issueCredential(String enryptedUin, String uinHash, String uinStatu
 			credStatus.setIdExpiryTimestamp(uinStatus.contentEquals(activeStatus) ? null : expiryTimestamp);
 			credStatus.setCreatedBy(IdRepoSecurityManager.getUser());
 			credStatus.setCrDTimes(DateUtils.getUTCCurrentDateTime());
-			if(enableConventionBasedId && (requestId != null)) {
```
Comment (LoganathanSekar7627) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java:748]: This should not be removed. This one is newly added, and this is not removed for resident.

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java [hunk 4]
```diff
@@ -785,41 +897,4 @@ protected byte[] convertToBytes(Object identity) throws IdRepoAppException {
 		}
 	}
 
-	private Map<String, HandleDto> checkAndGetHandles(IdRequestDTO request) throws IdRepoAppException {
```
Comment (LoganathanSekar7627) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java:788]: This should not be removed. This one is newly added, and this is not removed for resident.

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: id-repository/credential-request-generator/pom.xml [hunk 5]
```diff
@@ -7,11 +7,11 @@
 	<parent>
 		<groupId>io.mosip.idrepository</groupId>
 		<artifactId>id-repository-parent</artifactId>
-		<version>1.2.0.1</version>
+		<version>1.2.1.0</version>
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply id-repository/credential-request-generator/pom.xml:10]: Done

File: db_upgrade_scripts/mosip_idrepo/sql/1.2.0.1_to_1.2.1.0_upgrade.sql (hunk 1 above)
Comment (kameshsr) [reply db_upgrade_scripts/mosip_idrepo/sql/1.2.0.1_to_1.2.1.0_upgrade.sql:1]: https://github.com/mosip/id-repository/blob/0d56e166ff64713732a493262b63c5bf409cfcca/db_upgrade_scripts/mosip_idrepo/sql/1.1.5.5_to_1.2.0.1-B1_upgrade.sql#L104

it's already there in earlier upgrade script i.e. 1155 to 1201b1

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/provider/IdentityUpdateTrackerPolicyProvider.java (hunk 2 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/provider/IdentityUpdateTrackerPolicyProvider.java:21]: Done

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java (hunk 3 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java:748]: Added.

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java (hunk 4 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoServiceImpl.java:788]: Added

**Review by LoganathanSekar7627 — APPROVED** [review]

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/validator/IdRequestValidator.java [hunk 6]
```diff
@@ -241,8 +241,7 @@ public void validateRequest(Object request, Errors errors, String method) {
 						String schemaVersion;
 						if (requestMap.get(ROOT_PATH) != null) {
 							schemaVersion = String
-									.valueOf(((Map<String, Object>) requestMap.get(ROOT_PATH))
```
Comment (LoganathanSekar7627) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/validator/IdRequestValidator.java:244]: I think this is not related to resident change. Please keep it if it is there in release.

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/controller/IdRepoController.java [hunk 7]
```diff
@@ -402,7 +410,6 @@ public ResponseEntity<IdResponseDTO> updateAuthtypeStatus(
 					: getIdType(individualId).getIdType());
 			IdType idType = validator.validateIdType(authTypeStatusRequest.getIndividualIdType());
 			validator.validateIdvId(individualId, idType);
-			validator.validateAuthTypes(authTypeStatusRequest.getRequest());
```
Comment (LoganathanSekar7627) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/controller/IdRepoController.java:405]: I think this is not related to resident changes. Please keep it if it is there in release.

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoProxyServiceImpl.java [hunk 8]
```diff
@@ -449,9 +484,6 @@ protected byte[] getBiometricsForRequestedFormats(String uinHash, String fileNam
 		} catch (IdRepoAppUncheckedException e) {
 			mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_IMPL, "extractTemplate", e.getMessage());
 			throw new IdRepoAppException(e.getErrorCode(), e.getErrorText(), e);
-		} catch (InterruptedException e) {
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoProxyServiceImpl.java:452]: Done

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoProxyServiceImpl.java [hunk 9]
```diff
@@ -479,8 +511,7 @@ public IdResponseDTO updateIdentity(IdRequestDTO request, String uin) throws IdR
 				}
 				Uin uinObject=service.updateIdentity(request, uin);
 				mosipLogger.info("Uin updated");
-				String activeStatus = env.getProperty(ACTIVE_STATUS);
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/service/impl/IdRepoProxyServiceImpl.java:482]: Done

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/validator/IdRequestValidator.java (hunk 6 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/validator/IdRequestValidator.java:244]: Done

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/controller/IdRepoController.java (hunk 7 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/controller/IdRepoController.java:405]: Done

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/interceptor/IdRepoEntityInterceptor.java [hunk 10]
```diff
@@ -130,7 +132,7 @@ private <T extends UinInfo> void encryptDataOnSave(Serializable id, Object[] sta
 	public boolean onLoad(Object entity, Serializable id, Object[] state, String[] propertyNames, Type[] types) {
 		try {
 			List<String> propertyNamesList = Arrays.asList(propertyNames);
-			if (entity instanceof Uin || entity instanceof UinHistory || entity instanceof UinDraft) {
+			if (entity instanceof Uin || entity instanceof UinDraft) {
```
Comment (LoganathanSekar7627) [inline id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/interceptor/IdRepoEntityInterceptor.java:135]: Please check in this if it is present i release branch, add it back.

**Review by LoganathanSekar7627 — CHANGES_REQUESTED** [review]

File: id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/interceptor/IdRepoEntityInterceptor.java (hunk 10 above)
Comment (kameshsr) [reply id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/interceptor/IdRepoEntityInterceptor.java:135]: Done

**Review by LoganathanSekar7627 — APPROVED** [review]
